fix: NetworkVariables of enum types breaking the inspector [MTT-6212]

ShadauxCat opened the pull request:
fixes #2444 

## Changelog

- Fixed: Fixed the inspector throwing exceptions when attempting to render `NetworkVariable`s of enum types.

## Testing and Documentation

- No tests have been added.
- No documentation changes or additions were necessary.

ShadauxCat committed:
fix: NetworkVariables of enum types breaking the inspector [MTT-6212]

ShadauxCat committed:
Changelog

0xFA11 commented on com.unity.netcode.gameobjects/CHANGELOG.md:
```diff
@@ -17,6 +17,7 @@ Additional documentation and release notes are available at [Multiplayer Documen
 - Fixed issue where a client could throw an exception if abruptly disconnected from a network session with one or more spawned `NetworkObject`(s). (#2510)
 - Fixed issue where invalid endpoint addresses were not being detected and returning false from NGO UnityTransport. (#2496)
 - Fixed some errors that could occur if a connection is lost and the loss is detected when attempting to write to the socket. (#2495)
+- Fixed the inspector throwing exceptions when attempting to render `NetworkVariable`s of enum types. (#2529)
```
do we now render enum values in the inspector properly or do we just suppress/hide it? (didn't check the code yet :P but having it written in the changelog line would also be nice too, so I'm commenting this inline)

ShadauxCat replied:
They're rendered correctly. It's a fix to a regression, they're the way they were before the bug.

0xFA11 commented on com.unity.netcode.gameobjects/Editor/NetworkBehaviourEditor.cs:
```diff
@@ -81,7 +81,25 @@ private void RenderNetworkVariable(int index)
+                MethodInfo method;
+                if (isEquatable)
+                {
+                    method = typeof(NetworkBehaviourEditor).GetMethod("RenderNetworkContainerValueTypeIEquatable", BindingFlags.Public | BindingFlags.Instance | BindingFlags.FlattenHierarchy | BindingFlags.NonPublic);
```
can we use `nameof()` here for the constant string "RenderNetworkContainerValueTypeIEquatable" value?

ShadauxCat replied:
done

0xFA11 commented on com.unity.netcode.gameobjects/Editor/NetworkBehaviourEditor.cs:
```diff
@@ -81,7 +81,25 @@ private void RenderNetworkVariable(int index)
+                }
+                else
+                {
+                    method = typeof(NetworkBehaviourEditor).GetMethod("RenderNetworkContainerValueType", BindingFlags.Public | BindingFlags.Instance | BindingFlags.FlattenHierarchy | BindingFlags.NonPublic);
```
^ same for "RenderNetworkContainerValueType"

ShadauxCat replied:
done

0xFA11 reviewed (approved):
minor suggestions but other than that, all good

ShadauxCat committed:
- standards
- review feedback
- fixed an issue that could cause variables to render twice if the result of `NicifyVariableName()` is different from the actual variable name.

NoelStephensUnity reviewed (approved):
Tested under a couple of various conditions and it looks like that handles a couple of issues in one pass.
LGTM!

ShadauxCat committed:
Merge branch 'develop' into fix/networkvariable_enums_in_inspector
